### PR TITLE
FEAT-#4913: Enabling pyhdk

### LINKED ISSUE #4913: FEAT: Enabling pyhdk
We replace omniscidbe with pyhdk backend based on https://github.com/intel-ai/hdk

Docs update will be follow up

### PR DESCRIPTION
**Pull request by Garra1980**
## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4913 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by mvashishtha — DISMISSED** [review]
The files outside of docs/flow/modin/experimental/core/execution/native/implementations/omnisci_on_native/ LGTM.

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/pyhdk_worker.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    @classmethod
+    def setup_engine(cls):
+        """
+        Initialize PyHDK.
+
+        Do nothing if it is initiliazed already.
+        """
+        if cls._executor is None:
+            cls._config = pyhdk.buildConfig(**OmnisciLaunchParameters.get())
+            # cls._config = pyhdk.buildConfig()
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/omnisci_on_native/pyhdk_worker.py:43]: Should we remove this?

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module chooses a proper worker class."""
+
+try:
+    # Try to use PyHDK by default
+    from .pyhdk_worker import PyHdkWorker as DbWorker  # noqa
+except ModuleNotFoundError:  # fallback to omniscidbe4py package
+    from .omnisci_worker import OmnisciWorker as DbWorker  # noqa
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py:20]:
```suggestion
    from .omnisci_worker import OmnisciWorker as DbWorker  # noqa
    
__all__ = ["DbWorker"]

```

Comment (YarShev) [reply modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py:20]: Also, why do we need noqa here?

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module chooses a proper worker class."""
+
+try:
+    # Try to use PyHDK by default
+    from .pyhdk_worker import PyHdkWorker as DbWorker  # noqa
+except ModuleNotFoundError:  # fallback to omniscidbe4py package
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py:19]:
```suggestion
except ModuleNotFoundError:  # fallback to pyomniscidbe package
```
Which one exactly?

File: docs/flow/modin/experimental/core/execution/native/implementations/omnisci_on_native/index.rst [hunk 4]
```diff
@@ -244,15 +244,15 @@ by ``DFAlgNode`` based trees. Scalar computations are described by ``BaseExpr``
 * :doc:`Frame nodes <df_algebra>`
 * :doc:`Expression nodes <expr>`
 
-Interactions with OmniSci engine are done using ``OmnisciServer`` class. Queries use serialized
+Interactions with OmniSci engine are done using ``OmnisciWorker`` class. Queries use serialized
```
Comment (YarShev) [inline docs/flow/modin/experimental/core/execution/native/implementations/omnisci_on_native/index.rst:247]: Do we want to rename the `OmnisciOnNative` to `PyhdkOnNative` execution or something like that?

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py (hunk 3 above)
Comment (Garra1980) [reply modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py]: Fixed

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py (hunk 2 above)
Comment (Garra1980) [reply modin/experimental/core/execution/native/implementations/omnisci_on_native/db_worker.py]: Fixed

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/pyhdk_worker.py (hunk 1 above)
Comment (Garra1980) [reply modin/experimental/core/execution/native/implementations/omnisci_on_native/pyhdk_worker.py]: Fixed

**Review by YarShev — APPROVED** [review]

**Pull request merged**